### PR TITLE
Custom cookie validation

### PR DESCRIPTION
**Pull request by gltronred**
It seems that cookie-policy doesn't work and tests were removed.
I'm not very familiar with Apache http-components, so this solution can be not very good, please tell me, I will try to fix it.

This PR adds a new key, `:cookie-validation`, into the request. If this key is associated with a function, it is used as a `validate` method in proxy class extending `DefaultCookieSpec`. This proxy class is used instead of `BasicCookieSpec` which is returned by `cookie-spec` function

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(defproject clj-http "3.0.0-SNAPSHOT"
+(defproject lambdakazan/clj-http "2.0.4-SNAPSHOT"
```
Comment (dakrone) [inline project.clj]: Whoops, I don't think you meant to add this?

File: project.clj [hunk 2]
```diff
@@ -30,6 +30,11 @@
              :1.6 {:dependencies [[org.clojure/clojure "1.6.0"]]}}
   :aliases {"all" ["with-profile" "dev,1.5:dev,1.6:dev"]}
   :plugins [[codox "0.6.4"]]
+  :repositories [["snapshots" {:url "http://repo.lambdasoft.ru/artifactory/libs-snapshot-local"
+                               :sign-releases false
+                               :creds gpg}]
+                 ["releases" {:url "http://repo.lambdasoft.ru/artifactory/libs-release-local"
+                              :creds gpg}]]
```
Comment (dakrone) [inline project.clj]: Or these?

File: project.clj (hunk 1 above)
Comment (gltronred) [inline project.clj]: Sorry, I thought pull request will compare with a commit, not with a branch. Should I make another PR?

Comment (dakrone) [inline project.clj]: I'm sorry I missed this comment, another PR would work, or you can force-push to your branch and fix these
